src/inc/msquic.h

```diff
@@ -1161,6 +1161,7 @@ typedef enum QUIC_CONNECTION_EVENT_TYPE {
 #ifdef QUIC_API_ENABLE_PREVIEW_FEATURES
     QUIC_CONNECTION_EVENT_RELIABLE_RESET_NEGOTIATED         = 16,   // Only indicated if QUIC_SETTINGS.ReliableResetEnabled is TRUE.
     QUIC_CONNECTION_EVENT_ONE_WAY_DELAY_NEGOTIATED          = 17,   // Only indicated if QUIC_SETTINGS.OneWayDelayEnabled is TRUE.
+    QUIC_CONNECTION_EVENT_NETWORK_STATISTICS                = 18,
```

Comment:
The new event must be disabled by default, and opt in. Additionally, we'll need to a test to validate this behavior.